doc/developers/develop.rst

```diff
@@ -512,11 +512,15 @@ of estimators that allow programmatic inspection of their capabilities, such as
 sparse matrix support, supported output types and supported methods. The
 estimator tags are a dictionary returned by the method ``_get_tags()``. These
 tags are used by the common tests and the
-:func:`sklearn.utils.estimator_checks.check_estimator` function to decide what
-tests to run and what input data is appropriate. Tags can depend on estimator
-parameters or even system architecture and can in general only be determined at
-runtime. The default values for the estimator tags are defined in the
-``BaseEstimator`` class.
+:func:`~sklearn.utils.estimator_checks.check_estimator` function and
+:func:`~sklearn.utils.estimator_checks.parametrize_with_checks` decorator to
+decide what tests to run and what input data is appropriate. Tags can depend on
+estimator parameters or even system architecture and can in general only be
+determined at runtime. The default values for the estimator tags are defined in
+the :class:`~sklearn.base.BaseEstimator` class. If rolling your own estimator
+without inheriting from :class:`~sklearn.base.BaseEstimator`, you will need to
+implement the `_get_tags()` function. Besides, this function needs to at least
+return the tags defined below.
```

Comment:
It's not true, right ? given the use and definition of safe_tags you don't have to implement get_tags. Should we say "we highly recommend you to implement ..." ?

Reply:
I think one needs to implement `_get_tags()` if they don't inherit from BaseEstimator and if they don't want to rely on the default (which is likely).


However, I think we need to put all that after having presented `_more_tags()`.

Reply:
@jeremiedbb If you want to use tags in your third-party code without inheriting from `BaseEstimator`, you need to implement `_get_tags` in your estimator. Not implementing, it means that you don't care about `_get_tags` and in this case, we will internally default on something to run the checks with some default (which will be bad in 0.24 but hopefully better in 0.25/1.0 with API check only)

Reply:
okay but your MinimalClassifier does not inherit from BaseEstimator and does not implement get_tags :)

Reply:
Oh maybe I get the point of @jeremiedbb. The paragraph is stating about the `check_estimators` and `parametrize_with_checks`. In this case, we should probably mention that not implementing `_get_tags` and not inheriting will lead to using default tags in the checks.

Reply:
and later on add next to `_more_tags` that one should implement `_get_tags` to be able to use them in a class that does not inherit from `BaseEstimator`

Reply:
@jeremiedbb could you check the last commit to see if I got your point?

Reply:
Yes. Should we add in the second point (``* if your estimator does not inherit from``) that it's not recommended ? I'm not sure I remember correctly but I thought we wanted to say it's not recommended during the last meeting.

Reply:
I don't know if we mention this here or when we will rewrite the documentation of "Rolling your own estimator" with the two ways of doing and be more assertive in this regard.

Reply:
as you like